### PR TITLE
Widget builder as parameter

### PR DESCRIPTION
**Pull request by pinkeshmars**
### Widget builder as parameter

[Linear ticket](https://linear.app/flutterflow/issue/DEVR-601/docs-widget-builder-as-a-parameter) and [magic word](https://linear.app/docs/github#link-prs) Fixes DEVR-601 

## Type of change
- [ ] Typo fix 
- [x] New feature 
- [ ] Enhancement to current docs
- [ ] Removed outdated references
- [ ] Update assets

### CODE REVIEW

File: docs/resources/ui/components/using-components.md [hunk 1]
```diff
@@ -108,6 +108,145 @@ respective widgets and action triggers.
...
+            loading="lazy"
+            webkitAllowFullScreen
+            mozAllowFullScreen
+            allowFullScreen
+            allow="clipboard-write">
+        </iframe>
+    </div>
+    <p></p>
+
+2. Then, add the widget builder placeholder to the appropriate location within the component’s widget tree.
```
[the remark this answers is not in the document]
Comment (PoojaB26) [reply docs/resources/ui/components/using-components.md:157]: Use the suggestion or modify slightly if you feel so.

**Review by PoojaB26 — CHANGES_REQUESTED** [review]
Minor changes needed.

File: docs/resources/ui/components/using-components.md [hunk 2]
```diff
@@ -108,6 +108,145 @@ respective widgets and action triggers.
     </iframe>
 </div>
 
+### Widget Builder as Parameter
+
+The Widget Builder as Parameter feature allows you to create flexible and reusable UI components by passing widgets as parameters. This helps maintain a consistent design while allowing for customization when using the component in different places. By defining a base structure, you can easily reuse it with different content or styles across your app.
```
[the remark this answers is not in the document]
Comment (pinkeshmars) [reply docs/resources/ui/components/using-components.md:113]: Good point! updated.

File: docs/resources/ui/components/using-components.md [hunk 3]
```diff
@@ -108,6 +108,145 @@ respective widgets and action triggers.
...
+
+For example, say you’re building a custom dropdown component that displays a list of items. The structure of the dropdown remains the same, but the style and content of the dropdown items might differ. Without this feature, you would need to create multiple separate dropdown components for each style or content type. However, with this feature, you can pass the dropdown item as a parameter, making the dropdown component flexible enough to be used in various scenarios without creating new components every time.
```
[the remark this answers is not in the document]
Comment (pinkeshmars) [reply docs/resources/ui/components/using-components.md:115]: Updated!

File: docs/resources/ui/components/using-components.md [hunk 4]
```diff
@@ -108,6 +108,145 @@ respective widgets and action triggers.
...
+
+:::tip[possible use casses]
+- **Custom Cards**: Imagine you need to display product cards in an e-commerce app. You can build a reusable card component with parameters for the image, header, content, and call-to-action button. This card can be reused across multiple pages but with different content.
+- **Dynamic Forms**: Build a form component where different fields (TextFields, Dropdowns, or Checkboxes) are passed in as parameters. This allows you to reuse the same form structure but adapt to various input fields.
+- **Modular Layouts**: Create a consistent layout structure with areas like headers and footers that remain the same while passing in different body content as parameters to adapt to different pages.
+:::
+
+Let’s see how to use this feature with an example from an eComm app. On the shipping address page, you can maintain a consistent style and layout for input fields while allowing customization for the type of input—such as a TextField for the name, and a Dropdown for selecting country. Moreover, you can reuse the same component across other pages, such as the profile update.
```
[the remark this answers is not in the document]
Comment (pinkeshmars) [reply docs/resources/ui/components/using-components.md:123]: Done!

File: docs/resources/ui/components/using-components.md [hunk 5]
```diff
@@ -108,6 +108,145 @@ respective widgets and action triggers.
...
+
+![widget-builder-as-parameter-example.avif](imgs/widget-builder-as-parameter-example.avif)
```
[the remark this answers is not in the document]
Comment (pinkeshmars) [reply docs/resources/ui/components/using-components.md:125]: ok, updated!

File: docs/resources/ui/components/using-components.md [hunk 6]
```diff
@@ -108,6 +108,145 @@ respective widgets and action triggers.
...
+
+Here’s exactly how you do it:
+
+1. If you haven't already, start by creating a new component. Next, define a parameter and set its type to **Widget Builder**. To [pass data from the current component to the widget builder](#pass-parameter-from-component-to-widget-builder), you can specify a parameter within it.
```
[the remark this answers is not in the document]
Comment (pinkeshmars) [reply docs/resources/ui/components/using-components.md:129]: Used your suggestion!

File: docs/resources/ui/components/using-components.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (pinkeshmars) [reply docs/resources/ui/components/using-components.md:157]: Changed it slightly.

**Comment by pinkeshmars** [comment]
Hey @PoojaB26, all review comments are addressed now. Could you please check?

**Review by PoojaB26 — APPROVED** [review]
LGTM! Thanks for working on the changes!

**Pull request merged**